pkg/controller/deployment/operator_log_level.go

```diff
@@ -0,0 +1,37 @@
+package deployment
+
+import (
+	"strconv"
+
+	"k8s.io/component-base/logs"
+	"k8s.io/klog/v2"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+var operatorLogLevels = map[operatorv1.LogLevel]int{
+	operatorv1.Normal:   2,
+	operatorv1.Debug:    4,
+	operatorv1.Trace:    6,
+	operatorv1.TraceAll: 8,
+}
+
+// setOperatorLogLevel is used to set the value of klog verbosity
+// if current verbosity level does not match logLevel
+func setOperatorLogLevel(logLevel operatorv1.LogLevel) error {
+	if logLevel == "" {
+		logLevel = operatorv1.Normal
+	}
+
+	v := klog.Level(operatorLogLevels[logLevel])
+	// set verbosity by checking if V(n) and V(n+1) are enabled,
+	// change runtime verbosity level iff !V(n) or [V(n) and V(n+1)]
+	if !klog.V(v).Enabled() || (klog.V(v).Enabled() && klog.V(v+1).Enabled()) {
+		msg, err := logs.GlogSetter(strconv.Itoa(int(v)))
+		if err != nil {
+			return err
+		}
```

Comment:
Can this possible result in a non-nil error at all?

Reply:
I would say the chances are low but always good to have a defensive code path since in `logs.GlogSetter` they do already have an error case added:
https://github.com/openshift/cert-manager-operator/blob/ocpbugs-9994/vendor/k8s.io/component-base/logs/logs.go#L185-L193

Reply:
Updates in conversation thread beneath. This part of the implementation have been thoroughly changed in new commits.